openmdao/core/vec_wrapper.py

```diff
@@ -126,22 +129,49 @@ def _setup_get_funct(self, vecwrapper, meta, alloc_complex):
         # No unit conversion.
         # dparams vector does no unit conversion.
         if scale is None or vecwrapper.deriv_units:
-            if alloc_complex:
-                flatfunc = self._get_arr_complex
-                if is_scalar:
-                    func = self._get_scalar_complex
-                elif shapes_same:
-                    func = flatfunc
+
+            scaler = None
+            # Need the unknowns vector to unscale the value when any get is
+            # called during apply_linear, apply_nonlinear, or linearize. No
+            # other vectors return scaled values on get.
+            if vecwrapper.vectype == 'u':
+                scaler = meta.get('scaler')
+
+            if scaler:
+                if alloc_complex:
+                    flatfunc = self._get_arr_complex_scaler
```

Comment:
flatfunc is also returned at the end of the function, so it isn't a temporary ref.